src/twisted/internet/defer.py

```diff
@@ -1316,9 +1331,7 @@ def _cbDeferred(
             elif self.finishedCount == len(self.resultList):
                 # All the None values have been replaced by results, so we can
                 # safely cast away the Optional[].
```

Comment:
That comment is obsolete now that you're typecasting on the creation of `resultList` instead of casting it on completion.

Reply:
poof